umap/static/umap/js/umap.features.js

```diff
@@ -1,25 +1,70 @@
 U.FeatureMixin = {
   staticOptions: { mainColor: 'color' },
 
-  initialize: function (map, latlng, options) {
+  getSyncMetadata: function () {
+    return {
+      engine: this.map.sync,
+      subject: 'feature',
+      metadata: {
+        id: this.id,
+        layerId: this.datalayer?.id || null,
+        featureType: this.getClassName(),
+      },
+    }
+  },
+
+  onCommit: function () {
+    // When the layer is a remote layer, we don't want to sync the creation of the
+    // points via the websocket, as the other peers will get them themselves.
+    if (this.datalayer.isRemoteLayer()) return
+    const { subject, metadata, engine } = this.getSyncMetadata()
+    engine.upsert(subject, metadata, this.toGeoJSON())
+  },
+
+  getGeometry: function () {
+    return this.toGeoJSON().geometry
+  },
+
+  syncUpdatedProperties: function (properties) {
```

Comment:
Actually, because this `syncUpdatedProperties` function was only used once, I removed it and simplified the whole flow. I believe it's easier to follow now, and as a result it solves the naming problem we had 💇